src/context/todoProvider.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+  ReactNode, createContext, useContext, useEffect, useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+import { getTodos } from '../api/todos';
+
+type TodosProps = {
+  todos: Todo[];
+  taskName: string;
+  setTaskName: (query: string) => void;
+  filterBy: string;
+  setFilterBy: (query: string) => void,
+  count: number;
+  error: null | string;
+  setError: (err: string | null) => void;
+};
+
+const TodosContext = createContext<TodosProps>({
+  todos: [],
+  taskName: '',
+  setTaskName: () => undefined,
+  filterBy: 'all',
+  setFilterBy: () => undefined,
+  count: 0,
+  error: null,
+  setError: () => undefined,
+});
+
+type Props = {
+  children: ReactNode;
+};
+
+const dataFilter = (data: Todo[], filtr: string) => {
+  switch (filtr) {
+    case 'active':
+      return data.filter(task => !task.completed);
+    case 'completed':
+      return data.filter(task => task.completed);
+    default:
+      return data;
+  }
+};
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
```

Comment:
I prefer to separate todos and todos after applying filter, making one more state, for save previous todos state untouchable

but I will ask mentors about this